x/mongo/driver/topology/connection_options.go

```diff
@@ -136,14 +134,6 @@ func WithIdleTimeout(fn func(time.Duration) time.Duration) ConnectionOption {
 	}
 }
 
-// WithLifeTimeout configures the maximum life of a connection.
-func WithLifeTimeout(fn func(time.Duration) time.Duration) ConnectionOption {
-	return func(c *connectionConfig) error {
-		c.lifeTimeout = fn(c.lifeTimeout)
-		return nil
-	}
-}
-
```

Comment:
To check my understanding. Though this is removing public API from x, nobody should be using this option. Since it closes connections after this timeout regardless of use. I don't think anyone will be missing this.